### PR TITLE
Support of the WASM build

### PR DESCRIPTION
**Pull request by vhavlena**

### CODE REVIEW

File: 3rdparty/simlib/include/mata/simlib/util/shared_list.hh [hunk 1]
```diff
@@ -108,7 +108,7 @@ public:
 		}
 
 		if (elem)
-			--elem->counter_;
+			--elem->refcount_;
```
Comment (jurajsic) [inline 3rdparty/simlib/include/mata/simlib/util/shared_list.hh:111]: This seems like quite random fix, how did you find this bug?

**Review by jurajsic — APPROVED** [review]

File: 3rdparty/simlib/include/mata/simlib/util/shared_list.hh (hunk 1 above)
Comment (vhavlena) [reply 3rdparty/simlib/include/mata/simlib/util/shared_list.hh:111]: It simply does not compile. Maybe it has something to do with the emscripten that is used for the build (but the `counter_` is indeed not present there, it is a real error).

Comment (jurajsic) [reply 3rdparty/simlib/include/mata/simlib/util/shared_list.hh:111]: Really weird, maybe the function is never used, so the compiler removes it? But it should still not compile normally, weird.

**Review by Adda0 — APPROVED** [review]
This is great. Thank you for investigating.

**Pull request merged**